src/IssueViz.Security/Taint/TaintList/ViewModels/TaintIssuesControlViewModel.cs

```diff
@@ -164,9 +173,17 @@ private void ActiveDocumentTracker_OnDocumentFocused(object sender, ActiveDocume
         {
             activeDocumentFilePath = e.ActiveTextDocument?.FilePath;
             ApplyViewFilter(ActiveDocumentFilter);
+            UpdateServerType();
```

Comment:
@duncanp-sonar we're re-calculating server type in two places: ctor, and whenever an active document changes. This is under the assumption that if you change a solution, your active document will change. WDYT? 
Perhaps it's better to import `IActiveSolutionBoundTracker` and register to `SolutionBindingChanged`. WDYT?

cc @georgii-borovinskikh-sonarsource

Reply:
@georgii-borovinskikh-sonarsource so let's go with the cache 👍